### PR TITLE
Add bed placement

### PR DESCRIPTION
**Pull request by vurpo**
This makes bed placement possible. Beds get placed with the foot block on the clicked block, and the head block one block away from the foot block in the opposite direction from the player, just like in vanilla Minecraft. Placement with the head block in the air or inside another block is not possible.

### CODE REVIEW

File: src/main/java/net/glowstone/block/blocktype/BlockBed.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class BlockBed extends BlockType {
+
+    public BlockBed() {
+        setDrops(new ItemStack(Material.BED));
+    }
+
+    @Override
+    public void placeBlock(GlowPlayer player, GlowBlockState state, BlockFace face, ItemStack holding, Vector clickedLoc) {
+        BlockFace direction;
+        double rot = player.getLocation().getYaw() % 360;
```
Comment (EntityReborn) [inline src/main/java/net/glowstone/block/blocktype/BlockBed.java]: Stuff like this should probably be done in `getYaw()` to avoid duplicated/buggy code.

Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockBed.java]: Or add a method to `Location`: `getRotation` and document it returns degrees.

**Comment by turt2live** [comment]
Ticket: #119

**Comment by turt2live** [comment]
Looks good. Besides the obvious physics problems (out of scope), this seems ready to be pulled.

http://i.turt2live.com/WAlA.jpg

File: src/main/java/net/glowstone/block/blocktype/BlockBed.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            } else {
+                GlowServer.logger.warning("Placing " + getMaterial().name() + ": MaterialData was of wrong type (" + data.getClass().getName() + ")");
+            }
+        }
+    }
+
+    @Override
+    public void afterPlace(GlowPlayer player, GlowBlock block, ItemStack holding) {
+        if (block.getType() == Material.BED_BLOCK) {
+            BlockFace direction;
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockBed.java]: all of these variables shouldn't be declared and then set. Instead they should be set when declared.

**Comment by turt2live** [comment]
@vurp0 correction to previous comment. Please see diff.

**Comment by vurpo** [comment]
@turt2live Fixed that now.

**Comment by turt2live** [comment]
Sorry - bot fail :(

**Comment by turt2live** [comment]
Appears to work, thanks!

**Comment by SpaceManiac** [comment]
Merged in 7dae0f0ff64146b54821e417bbf2024d221d299b.

**Pull request closed without merging**